docs/generator_template.md

```diff
@@ -0,0 +1,569 @@
+5. Create a new file named `index.js` in your **template** directory. This file is used to define the **logic** for your template.
+6. Create a new directory named `partials` in your template directory. This directory is used to store the **partial HTML** files that are used to create the final documentation.
+7. Create a new directory named `assets` in your template directory. This directory is used to **store the CSS and JavaScript files that are used to style the final documentation or website**.
+
+Lets break it down:
+
+### `package.json` file
+
+The `package.json` file is used to define the dependencies for your template. Here's an example `package.json` file:
+
+``` json
+{
+  "name": "python-mqtt-client-template",
+  "version": "0.0.1",
+  "description": "A template that generates a Python MQTT client using MQTT.",
+  "generator": {
+    "renderer": "react",
+    "apiVersion": "v1",
+    "generator": ">=1.10.0 <2.0.0",
+    "supportedProtocols": ["mqtt"]
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+
+Here's what is contained in the code snippet above:
+
+- **name** -the name of your template.
+- **version** - the current version of your template.
+- **description** - a description of what your template does.
+- **generator** - specify generator [specific configuration](https://www.asyncapi.com/docs/tools/generator/configuration-file).
+  - **renderer** - can either be `react` or `nunjucks`. In this case the generator will pass your template to the react render engine to generate the output.
+  - **apiVersion** - specifies which major version of the [Parser-API](https://github.com/asyncapi/parser-api) your template will use.
+  - **generator** - a string representing the generator version-range your template is compatible with.
+  - **supportedProtocols** - A list that specifies which protocols are supported by your template.
+- **dependencies** - specifies which version of [`@asyncapi/generator-react-sdk`](https://github.com/asyncapi/generator-react-sdk) should be used.
+
+Run the command `npm install` on your terminal to install the dependencies specified in `package.json`.
+
+### `index.js` file
+
+The index.js file is used to define the logic for your template. Inside the template folder, create an `index.js` file and add the code snippet below:
+
+```js
+//1
+import { File } from '@asyncapi/generator-react-sdk'
+//2
+export default function ({ asyncapi }) {
+  //3
+  return <File name="client.py">{asyncapi.info().title()}</File>
+}
+```
+
+The code snippet above does the following:
+
+1. Import the `generator-react-sdk` dependency.
+2. The `asyncapi` argument is an instance of the [AsyncAPI Parser](https://www.asyncapi.com/docs/tools/generator/parser). It will allow you to access the content of the asyncapi document in your template using helper functions.
```

Comment:
```suggestion
2. The `asyncapi` argument is an instance of the [AsyncAPI Parser](https://www.asyncapi.com/docs/tools/generator/parser). It will allow you to access the content of the AsyncAPI document in your template using helper functions.
```